setup.ts

```diff
@@ -404,6 +405,107 @@ async function importData(): Promise<void> {
     );
   }
 }
+async function verifySmtpConnection(
+  config: Record<string, string>
+): Promise<boolean> {
+  const transporter = nodemailer.createTransport({
+    host: config.SMTP_HOST,
+    port: Number(config.SMTP_PORT),
+    secure: config.SMTP_SSL_TLS === "true",
+    auth: {
+      user: config.SMTP_USERNAME,
+      pass: config.SMTP_PASSWORD,
+    },
+  });
+
+  try {
+    await transporter.verify();
+    console.log("SMTP connection verified successfully.");
+    return true;
+  } catch (error: any) {
+    console.error("SMTP connection verification failed:");
+    return false;
+  } finally {
+    transporter.close();
+  }
+}
+async function configureSmtp(): Promise<void> {
+  console.log(
+    "SMTP Configuration is necessary for sending Emails through Talawa"
+  );
+  const { shouldConfigureSmtp } = await inquirer.prompt({
+    type: "confirm",
+    name: "shouldConfigureSmtp",
+    message: "Would you like to configure SMTP for Talawa to send emails?",
+    default: true,
+  });
+
+  if (!shouldConfigureSmtp) {
+    console.log("SMTP configuration skipped.");
+    return;
+  }
+
+  const smtpConfig = await inquirer.prompt([
+    {
+      type: "input",
+      name: "SMTP_HOST",
+      message: "Enter SMTP host:",
+    },
+    {
+      type: "input",
+      name: "SMTP_PORT",
+      message: "Enter SMTP port:",
+    },
+    {
+      type: "input",
+      name: "SMTP_USERNAME",
+      message: "Enter SMTP username:",
+    },
+    {
+      type: "password",
+      name: "SMTP_PASSWORD",
+      message: "Enter SMTP password:",
+    },
+    {
+      type: "confirm",
+      name: "SMTP_SSL_TLS",
+      message: "Use SSL/TLS for SMTP?",
+      default: false,
+    },
+  ]);
+
+  const isValidSmtpConfig =
+    smtpConfig.SMTP_HOST &&
+    smtpConfig.SMTP_PORT &&
+    smtpConfig.SMTP_USERNAME &&
+    smtpConfig.SMTP_PASSWORD;
+
+  if (!isValidSmtpConfig) {
+    console.error(
```

Comment:
There can be multiple reason for failuer. Pleas add error in the log and also make sure not to log password if exists in error

Reply:
Sure

Reply:
@sumitra19jha 
Added error message in the log

https://github.com/PalisadoesFoundation/talawa-api/assets/116940083/c774b6ad-9721-42ca-bcad-561c45e731e4